src/Findologic/Config/FindologicConfigService.php

```diff
@@ -0,0 +1,307 @@
+                $merged[$cur->getConfigurationKey()] = $cur->getConfigurationValue();
+            }
+        }
+
+        return $merged;
+    }
+
+    /**
+     * @param array|bool|float|int|string|null $value
+     */
+    public function set(string $key, $value, string $salesChannelId, string $languageId): void
+    {
+        // reset internal cache
+        $this->configs = [];
+
+        $key = trim($key);
+        $this->validate($key, $salesChannelId, $languageId);
+
+        $id = $this->getId($key, $salesChannelId, $languageId);
+        if ($value === null) {
+            if ($id) {
+                $this->finSearchConfigRepository->delete([['id' => $id]], Context::createDefaultContext());
+            }
+
+            return;
+        }
+
+        $data = [
+            'id' => $id ?? Uuid::randomHex(),
+            'configurationKey' => $key,
+            'configurationValue' => $value,
+            'salesChannelId' => $salesChannelId,
+            'languageId' => $languageId,
+        ];
+        $this->finSearchConfigRepository->upsert([$data], Context::createDefaultContext());
+    }
+
+    public function delete(string $key, string $salesChannel, string $languageId): void
+    {
+        $this->set($key, null, $salesChannel, $languageId);
+    }
+
+    private function validate(string $key, string $salesChannelId, string $languageId): void
+    {
+        $key = trim($key);
+        if ($key === '') {
+            throw new InvalidKeyException('key may not be empty');
+        }
+        if ($salesChannelId && !Uuid::isValid($salesChannelId)) {
+            throw new InvalidUuidException($salesChannelId);
+        }
+        if ($languageId && !Uuid::isValid($languageId)) {
+            throw new InvalidUuidException($languageId);
+        }
+    }
+
+    private function getId(string $key, string $salesChannelId, string $languageId): ?string
+    {
+        $criteria = $this->buildCriteriaWithKey($key, $salesChannelId, $languageId);
+        $ids = $this->finSearchConfigRepository->searchIds($criteria, Context::createDefaultContext())->getIds();
+
+        return array_shift($ids);
+    }
+
+    private function buildCriteriaWithKey(string $key, string $salesChannelId, string $languageId): Criteria
+    {
+        $criteria = new Criteria();
+        $criteria->addFilter(
+            new EqualsFilter('configurationKey', $key),
+            new EqualsFilter('salesChannelId', $salesChannelId),
+            new EqualsFilter('languageId', $languageId)
+        );
+
+        return $criteria;
+    }
+
+    private function load(string $salesChannelId, string $languageId): array
+    {
+        $key = sprintf('%s-%s', $salesChannelId, $languageId);
+
+        if (isset($this->configs[$key])) {
+            return $this->configs[$key];
+        }
+
+        $criteria = new Criteria();
+        if (method_exists($criteria, 'setTitle')) {
```

Comment:
Why are we checking if this method exists, and what does it?

Reply:
No idea, but it was introduced in later Shopware version and was failing tests for older versions.

Reply:
So the error was about not setting it? Could you send me a link to the GitHub Actions build where this is failing?

Reply:
Not it was that the method was undefined or did not exist. Yeah sure let me check, it was I think maybe 6.1. Will check.

Reply:
Did you find the related build?

Reply:
I will check this myself. Not relevant right now.